R-package/R/xgb.plot.deepness.R

```diff
@@ -1,155 +1,149 @@
+#' are plotted on top of each other:
 #' \itemize{
-#'  \item Count: number of leaf per level of deepness;
-#'  \item Weighted cover: noramlized weighted cover per leaf (weighted number of instances).
+#'  \item the distribution of the number of leafs in a tree model at a certain depth;
+#'  \item the distribution of average weighted number of observations ("cover") 
+#'        ending up in leafs at certain depth.
 #' }
+#' Those could be helpful in determining sensible ranges of the \code{max_depth} 
+#' and \code{min_child_weight} parameters.
+#' 
+#' When \code{which="max.depth"} or \code{which="med.depth"}, plots of either maximum or median depth
+#' per tree with respect to tree number are created. And \code{which="med.weight"} allows to see how
+#' a tree's median absolute leaf weight changes through the iterations.
 #'
-#' This function is inspired by the blog post \url{http://aysent.github.io/2015/11/08/random-forest-leaf-visualization.html}
+#' This function was inspired by the blog post
+#' \url{http://aysent.github.io/2015/11/08/random-forest-leaf-visualization.html}.
+#' 
+#' @return
+#' 
+#' Other than producing plots (when \code{plot=TRUE}), the \code{xgb.plot.deepness} function
+#' silently returns a processed data.table where each row corresponds to a terminal leaf in a tree model,
+#' and contains information about leaf's depth, cover, and weight (which is used in calculating predictions).
+#' 
+#' The \code{xgb.ggplot.deepness} silently returns either a list of two ggplot graphs when \code{which="2x1"}
+#' or a single ggplot graph for the other \code{which} options.
 #'
+#' @seealso 
+#' 
+#' \code{\link{xgb.train}}, \code{\link{xgb.model.dt.tree}}.
+#' 
 #' @examples
+#' 
 #' data(agaricus.train, package='xgboost')
 #'
 #' bst <- xgboost(data = agaricus.train$data, label = agaricus.train$label, max_depth = 15,
-#'                  eta = 1, nthread = 2, nrounds = 30, objective = "binary:logistic",
-#'                  min_child_weight = 50)
+#'                eta = 0.1, nthread = 2, nrounds = 50, objective = "binary:logistic",
+#'                subsample = 0.5, min_child_weight = 2)
 #'
-#' xgb.plot.deepness(model = bst)
+#' xgb.plot.deepness(bst)
+#' xgb.ggplot.deepness(bst)
+#' 
+#' xgb.plot.deepness(bst, which='max.depth', pch=16, col=rgb(0,0,1,0.3), cex=2)
+#' 
+#' xgb.plot.deepness(bst, which='med.weight', pch=16, col=rgb(0,0,1,0.3), cex=2)
 #'
+#' @rdname xgb.plot.deepness
 #' @export
-xgb.plot.deepness <- function(model = NULL) {
-  if (!requireNamespace("ggplot2", quietly = TRUE)) {
-    stop("ggplot2 package is required for plotting the graph deepness.",
-         call. = FALSE)
-  }
-  
-  if (!requireNamespace("igraph", quietly = TRUE)) {
-    stop("igraph package is required for plotting the graph deepness.",
-         call. = FALSE)
-  }
+xgb.plot.deepness <- function(model = NULL, which = c("2x1", "max.depth", "med.depth", "med.weight"),
```

Comment:
So far all the plotting functions have the same prefix `xgb.plot`. This change doesn't need to change the function for the sake of compatibility of the existing user scripts.

Can you rename  `xgb.plot.deepness` as `xgb.deepness`, and please also rename `xgb.ggplot.deepness` as `xgb.plot.deepness`? If you want to keep the name of `xgb.ggplot.deepness`, you can add a function `xgb.plot.deepness` to simply call `xgb.ggplot.deepness`, so that even somehow we need to remove ggplot2 from xgboost, we don't need to change the exposed function name.

My opinion holds for `xgb.plot.importance` as well.